wallet/test/Test/Pos/Wallet/Web/Methods/PaymentSpec.hs

```diff
@@ -58,10 +71,19 @@ oneNewPaymentSpec = walletPropertySpec oneNewPaymentDesc $ do
     srcAddr <- getAddress srcAccId
     -- Dunno how to get account's balances without CAccModifier
     initBalance <- getBalance srcAddr
+    let topBalance = unsafeGetCoin initBalance `div` 2
     -- `div` 2 to leave money for tx fee
-    coins <- pick $ mkCoin <$> choose (1, unsafeGetCoin initBalance `div` 2)
-    void $ lift $ newPayment pswd srcAccId dstCAddr coins def
-    dstAddr <- lift $ decodeCTypeOrFail dstCAddr
+    -- FIXME: this might be inafficient
```

Comment:
we should write a function which split `K` into `k` parts and put it to `Pos.Util`. it's used in several places

Reply:
Can you be more specific?

I could write this as 
```
coins <- pick $ vectorOf destLen $ mkCoin <$> choose (1, unsafeGetCoin initBalance `div` (2*destLen))
```

where `destLen` is `n` and `initBalance` is `K` in your case. This would be a bit conservative as each output would contain max `K/(2*n)` coins. 

If we go with:
```
coins <- pick $ vectorOf destLen $ mkCoin <$> choose (1, unsafeGetCoin initBalance `div` destLen)
```

Then each output would contain max `K/n` coins and in this case we might trigger `Insufficient founds` error because we won't have enough money to count in for fees.

With:
```
let topBalance = unsafeGetCoin initBalance `div` 2
 coins <- pick $ suchThat (vectorOf destLen $ mkCoin <$> choose (1, topBalance)) ((< topBalance) . fromIntegral . sumCoins)
```

each output would contain max `K/2` coins so it increases search space (at least if some bug is triggered by bigger amounts being sent). Not sure how important it is but search space is a bit bigger.

> we should write a function which split K into n parts and put it to Pos.Util. it's used in several places

So my conclusion is that you thing we should have something like:
```
coinPart coin n = mkCoin $ unsafeGetCoin coin `div` n
```

am I right?

(I am not aware of other places - so maybe pointing this out would understand better what you meant)

Reply:
If we have a function which split `K` into `n` parts, then we can choose `K <- choose (n, topBalance)` and then call this function

Reply:
ok, I think I got it. Let me try and ping you back